### PR TITLE
Using Promises: Clarify guarantees (#798); related light touches

### LINKED ISSUE #798: Issue with "Using Promises": …
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises

#### What information was incorrect, unhelpful, or incomplete?
The following line: 

> Callbacks added with `then()`, as above, will be called even _after_ the success or failure of the asynchronous operation.

Is somewhat unclear to me. The word "even" suggests surprise in the author's tone, like there is some other outcome we should be expected to see outside of "success or failure", but "success or failure" accounts for 100% of all possible outcomes. In combination with the emphasis on _after_, I'm also not too sure how to read this in my mind.

#### Specific section or headline?
Guarantees

#### What did you expect to see?
This is how I understand this guarantee in slightly modified wording:

> Callbacks added with `then()`, as above, will be always be called _after_ the success or failure of the asynchronous operation (as opposed to passed-in callbacks, where the callbacks my be called at any point during the asynchronous operation).

#### Did you test this? If so, how?
N/A

I hope that makes sense, thanks in advance!

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/guide/using_promises`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises
* GitHub URL: https://github.com/mdn/content/blob/master/files/en-us/web/javascript/guide/using_promises/index.html
* Report started: 2020-12-31T08:59:31.941Z

</details>

### PR DESCRIPTION
**Pull request by danburzo**
This PR attempts to clarify the _Promise guarantees_ statements (see #798) with very light touches. I've also simplified the intro somewhat, and fixed a broken link.

(Please let me know if you'd prefer PRs to be laser-focused on particular issues, or whether I can continue to lump in minor changes with them!)

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]
Very well written and explained — it’s great to get refinements made at this level of detail and quality, and on something so fundamental and important for devs to understand.

**Comment by sideshowbarker** [comment]
Thanks much, and congrats on landing your first patch here — welcome aboard 🎉

**Comment by danburzo** [comment]
Thanks! 🙌

**Comment by sideshowbarker** [comment]
> (Please let me know if you'd prefer PRs to be laser-focused on particular issues, or whether I can continue to lump in minor changes with them!)

I won’t attempt to assert I’m speaking for others here, but what I can say is that I’m not aware us having any policy that discourages slipping in other spot fixes. And I know I’ve done it myself at least.

(And for this project, given the long-tail disparate set of contributors I think we’re likely to end up with — and that we should try to attract — trying to enforce a policy of laser-focus in patches would both be futile in practice and also counter-productive, in that it would just be annoying to contributors while also making it more likely that minor changes just would get made (because it’s more work for contributors to do extra follow-up PRs, so those are less likely to happen).

So in other words, I think you can feel OK with going ahead and fixing whatever other problems you happen to notice when working on a patch for the actual important thing you want to get changed.

That said, I think as with other projects, it’s probably just a case of “let common sense prevail” — if there are so many other spot fixes in a patch that the volume of them obscures the actually-important change being made, then that’d be time to consider moving the spot fixes to a separate follow-up PR.

**Comment by danburzo** [comment]
Thank you for the detailed response, sounds good to me! I've also taken note of the _(fixes #IssueID)_ vs. simple _(#IssueID)_ convention.
